### PR TITLE
Don't crash if there's no package list client

### PR DESCRIPTION
**Pull request by jhrozek**
# Summary

If a github App provider is enrolled but no fallback token configured through `GitHubAppConfig.FallbackClient` then the github config structure's `packageListingClient` will be `nil`. While this only affects the initial reconcile of the artifacts on repo registration and not retrieving the artifacts through webhook notifications, minder shouldn't crash.

The patch also checks the result for null because on error the result might be null but the err handler was dereferencing result unconditionally.

## Change Type

***Mark the type of change your PR introduces:***

- [x] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

enroll with a github app without the fallback client

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: internal/providers/github/common.go [hunk 1]
```diff
@@ -176,6 +181,10 @@ func (c *GitHub) ListPackagesByRepository(
 		var resp *github.Response
 		var err error
 
+		if c.packageListingClient == nil {
```
Comment (puerco) [inline internal/providers/github/common.go:184]: I think the condition you're introducing on line 170 makes this code unreachable.

File: internal/providers/github/common.go [hunk 2]
```diff
@@ -167,6 +167,11 @@ func (c *GitHub) ListPackagesByRepository(
 	// create a slice to hold the containers
 	var allContainers []*github.Package
 
+	if c.packageListingClient == nil {
+		zerolog.Ctx(ctx).Error().Msg("No client available for listing packages")
+		return allContainers, nil
```
Comment (puerco) [inline internal/providers/github/common.go:172]: Should this return an error instead?

File: internal/providers/github/common.go [hunk 3]
```diff
@@ -167,6 +167,11 @@ func (c *GitHub) ListPackagesByRepository(
 	// create a slice to hold the containers
 	var allContainers []*github.Package
 
+	if c.packageListingClient == nil {
```
Comment (JAORMX) [inline internal/providers/github/common.go:170]: @jhrozek for my education, in what case would there be no package listing client?

File: internal/providers/github/common.go (hunk 1 above)
Comment (jhrozek) [reply internal/providers/github/common.go]: thanks, I started fixing the code where I saw the crash happening and then didn't remove this condition.

File: internal/providers/github/common.go (hunk 2 above)
Comment (jhrozek) [reply internal/providers/github/common.go]: good point, I changed the code to return an error code from the GH function, but then catch and ignore the error with a log message on the only place where it's used.

File: internal/providers/github/common.go (hunk 3 above)
Comment (jhrozek) [reply internal/providers/github/common.go:176]: > @jhrozek for my education, in what case would there be no package listing client?

I added an explanation to the PR description.

**Review by eleftherias — APPROVED** [review]

**Review by JAORMX — APPROVED** [review]

**Pull request merged**